### DISCUSSION TITLE
typescript only displays diagnostics for current buffer

### DISCUSSION THREAD

**Discussion by andrewbanchich** [joaotavora/eglot#1350] [Q&A] [answered]
I can't seem to figure out how to get a TypeScript project to display diagnostics for the whole project. When I open various files, I only see flymake diagnostics for that single file.

I thought maybe there was an issue with the project root, but `M-: (project-current)` displays the correct root.

How can I get eglot to display diagnostics for every file in a TypeScript project?

**Comment by joaotavora** [comment]
I'm not looking at your screen, so I don't know where you are looking for "diagnostics for the whole project", but the command `M-x flymake-show-project-diagnostics` is how `flymake.el` hooks to `project.el`.  Eglot tries to contribute, but not all servers support this.  Try it.

**Comment by andrewbanchich** [comment]
I actually just realized this is happening to my Rust projects too now, so it's not just my typescript setup. I looked into this a bit and it looks like there was [an update to consult-flymake](https://github.com/minad/consult/pull/589) at some point in time that changed how it works, but I'm unclear on some things about it. I'll open a discussion there.

Thanks!

**Reply by joaotavora** [reply]
I'm closing this, as it's hard to have this dicussion be useful without any hint of what you are doing with your Emacs.  It's impossible to tell how where you expect Emacs to "display diagnostics for the whole project".  

The way _I_ know and support to see all diagnostics for a project is via `M-x flymake-show-project-diagnostics`. As far as I can tell, it works fine in my C++ and Python projects, for example. I don't know any consult things.  I'm happy if they work for you, but that's indeed something you need to clarify over there.

**Reply by andrewbanchich** [reply]
Sorry, I had reopened and was about to post more context / information but got pulled away.

**Comment by andrewbanchich** [comment]
I'm experiencing several issues with flymake diagnostics. I'm using eglot as well as `consult-flymake` to jump to the next error.

1. When I call `consult-flymake` it only displays diagnostics for the current buffer. This was a combo of a deliberate API change from `consult-flymake` and [a current bug](https://github.com/minad/consult/discussions/919). Doesn't seem related to `eglot` at all, just adding here for context.
2. When I run `flymake-show-project-diagnostics` in a typescript project, it shows only diagnostics for files I have opened. If I run it in a Rust project, it shows diagnostics for all files, even if I haven't opened them.
3. The modeline stats for flymake diagnostics only display the current buffer's stats, regardless of whether its Rust or TypeScript.

**Reply by joaotavora** [reply, marked as answer]
1. OK, I hope they fix it.  I've had a brief look at your linked issues.  If they need Flymake to provide some kind of API for the info it uses in `flymake-show-project-diagnostics`, that can probably be done.
2. That's how it works.  It depends completely on the language server and your server-specific configuration (read about this in the manual) if it gives you diagnostics for files not yet visited.
3. That's how it should work. Maybe we could augment it to allow the user to configure project-wide counts. Make a feature request.

**Reply by andrewbanchich** [reply]
Ah, gotcha. Thanks for the explanation!
